source/common/http/conn_pool_grid.cc

```diff
@@ -325,13 +325,13 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
   } else {
     // Before skipping to the next pool, make sure it has been created.
     createNextPool();
-    ++pool;
+    pool = http2_pool_.get();
   }
   auto wrapped_callback =
-      std::make_unique<WrapperCallbacks>(*this, decoder, pool, callbacks, overriding_options);
+      std::make_unique<WrapperCallbacks>(*this, decoder, callbacks, overriding_options);
   ConnectionPool::Cancellable* ret = wrapped_callback.get();
   LinkedList::moveIntoList(std::move(wrapped_callback), wrapped_callbacks_);
-  if (wrapped_callbacks_.front()->newStream() == StreamCreationResult::ImmediateResult) {
+  if (wrapped_callbacks_.front()->newStream(pool) == StreamCreationResult::ImmediateResult) {
```

Comment:
This predates you, but might as well use `ret` instead of `wrapped_callbacks_.front()`.

Reply:
done